ethcore/src/service.rs

```diff
@@ -130,12 +132,16 @@ impl IoHandler<NetSyncMessage> for ClientIoHandler {
 	}
 
 	#[cfg_attr(feature="dev", allow(single_match))]
-	fn message(&self, io: &IoContext<NetSyncMessage>, net_message: &NetSyncMessage) {
-		if let UserMessage(ref message) = *net_message {
+	fn message(&self, io: &IoContext<NetSyncMessage>, net_message: &mut NetSyncMessage) {
```

Comment:
this is a bit horrible - we don't know someone won't want to use it after the call.

couldn't it be passed by move to guarantee correctness?